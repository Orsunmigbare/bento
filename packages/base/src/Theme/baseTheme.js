import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
