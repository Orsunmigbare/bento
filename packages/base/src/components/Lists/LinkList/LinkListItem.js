import React from "react"
import PropTypes from "prop-types"
import styled, { css } from "styled-components"

const Caret = ({ className }) => (
  <svg className={className} width="13" height="19" viewBox="0 0 13 19">
    <path d="M2.8 0L0 2.8l6.6 6.6L0 16l2.8 2.8 9.4-9.4L2.8 0z" />
  </svg>
)

Caret.propTypes = { className: PropTypes.string }
Caret.defaultProps = { className: "" }

const LinkListItem = ({ className, children, onClick, to }) => (
  <ListItem className={className}>
    <Button onClick={onClick && onClick} href={to}>
      {children} <Cursor />
    </Button>
  </ListItem>
)

LinkListItem.defaultProps = {
  className: "",
  defaultProps: null
}

LinkListItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func
}

const Cursor = styled(Caret)`
  fill: ${color.line};
  margin-left: ${spacing.small};
`

const ListItem = styled.li`
  list-style: none;
  border-bottom: 1px solid ${color.line};
  position: relative;
  &:last-child {
    border-bottom: none;
  }
`

const ButtonStyle = css`
  text-decoration: none;
  color: black;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  padding: ${spacing.small}
    ${spacing.medium};
  display: block;
  font-weight: normal;
  display: flex;
  justify-content: space-between;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  cursor: pointer;
`

const Button = styled.a`
  ${ButtonStyle};
  &:hover,
  &:focus {
    ${Cursor} {
      fill: ${color.primary};
    }
  }
`

export default LinkListItem
