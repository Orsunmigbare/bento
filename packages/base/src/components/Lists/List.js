import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
`

const List = ({ children, className }) => (
  <ListUl className={className}>{children}</ListUl>
)

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element
  ]).isRequired,
  className: PropTypes.string
}

export default List
