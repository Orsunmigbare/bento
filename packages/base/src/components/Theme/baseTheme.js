export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
