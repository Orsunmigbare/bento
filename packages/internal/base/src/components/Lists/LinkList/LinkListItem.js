import React from "react"
import PropTypes from "prop-types"
import styled, { css } from "styled-components"
import CaretRight from "../../Icons/CaretRight"
import { themePropTypes } from "../../../constants/themeContants"
import {
  spacing,
  fontWeight,
  fontFamily,
  font,
  color,
  applyVariants
} from "../../../theming"
import ThemeComponent from "../../Theme/ThemeComponent"

const themeProps = {
  iconComponent: {
    name: "COMPONENT_LINK_LIST_ITEM_ICON",
    description: "",
    type: themePropTypes.component
  },
  icon: {
    name: "LINK_LIST_ITEM_CARET",
    description: "",
    type: themePropTypes.style
  },
  listItem: {
    name: "LINK_LIST_ITEM",
    description: "",
    type: themePropTypes.style
  },
  itemButton: {
    name: "LINK_LIST_ITEM_BUTTON_STYLE",
    description: "",
    type: themePropTypes.style
  },
  link: {
    name: "LINK_LIST_ITEM_BUTTON",
    description: "",
    type: themePropTypes.style
  },
  hover: {
    name: "LINK_LIST_ITEM_BUTTON_HOVER_ICON",
    description: "",
    type: themePropTypes.style
  }
}

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={themeProps.iconComponent}
    fallback={CaretRight}
    {...props}
  />
)

const LinkListItem = ({ className, children, onClick, to }) => (
  <ListItem className={className}>
    <Button onClick={onClick && onClick} href={to}>
      {children} <Cursor />
    </Button>
  </ListItem>
)

LinkListItem.defaultProps = {
  className: ""
}

LinkListItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func
}

LinkListItem.themeProps = themeProps

const Cursor = styled(IconComponent)`
  fill: ${color.wcag};
  position: absolute;
  right: ${spacing.small};
  ${applyVariants(themeProps.icon)};
`

const ListItem = styled.li`
  list-style: none;
  border-bottom: 1px solid ${color.line};
  position: relative;
  &:last-child {
    border-bottom: none;
  }
  ${applyVariants(themeProps.listItem)};
`

const ButtonStyle = css`
  text-decoration: none;
  color: black;
  font-size: ${font.base};
  font-family: ${fontFamily.body()};
  padding: ${spacing.small} ${spacing.large} ${spacing.small} ${spacing.medium};
  display: block;
  font-weight: ${fontWeight.normal};
  display: flex;
  justify-content: space-between;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  cursor: pointer;
  ${applyVariants(themeProps.itemButton)};
`

const Button = styled.a`
  ${ButtonStyle};
  &:hover,
  &:focus {
    ${Cursor} {
      fill: ${color.primary};
      ${applyVariants(themeProps.hover)};
    }
  }
  ${applyVariants(themeProps.link)};
`

export default LinkListItem
