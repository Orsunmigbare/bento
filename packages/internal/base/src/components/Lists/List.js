import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { themify, themeProps } from "@staccx/theme"

const List = ({ children, className, ...restProps }) => (
  <ListUl className={className} {...restProps}>
    {children}
  </ListUl>
)

export const LIST = "LIST"

export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${themify(LIST)};
`

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string
}

export default List
