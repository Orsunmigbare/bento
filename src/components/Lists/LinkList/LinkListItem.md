```jsx
<List>
  <LinkListItem to="#">Lenke til greie</LinkListItem>
  <LinkListItem to="#">Annen lenke her</LinkListItem>
</List>
```
