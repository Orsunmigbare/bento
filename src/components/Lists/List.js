import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, commonPropTypes, themePropTypes } from "../../theming"

const List = ({ children, className, ordered, ...restProps }) => {
  if (!ordered) {
    return (
      <ListUl className={className} {...restProps}>
        {children}
      </ListUl>
    )
  }
  return (
    <ListOl className={className} {...restProps}>
      {children}
    </ListOl>
  )
}

List.themeProps = {
  list: {
    name: "LIST",
    description: "",
    type: themePropTypes.style
  }
}
export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${applyVariants(List.themeProps.list)};
`

export const ListOl = ListUl.withComponent("ol")

List.defaultProps = {
  className: "",
  ordered: false
}

List.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  /**
   * Prop to set ordered or unordered List (ul/ol)
   */
  ordered: PropTypes.bool
}

export default List
