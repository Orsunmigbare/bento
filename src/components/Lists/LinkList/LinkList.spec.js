import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import LinkListItem from "./LinkListItem"
import List from "../List.js"

describe("LinkListItem", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <LinkListItem>One ListElement</LinkListItem>
              <LinkListItem>Another List Element</LinkListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <LinkListItem className="className">One ListElement</LinkListItem>
              <LinkListItem className="className">
                Another List Element
              </LinkListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("to", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <LinkListItem to="/">One ListElement</LinkListItem>
              <LinkListItem to="/">Another List Element</LinkListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("onClick", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <LinkListItem onClick={e => console.log(e.target.value)}>
                One ListElement
              </LinkListItem>
              <LinkListItem onClick={e => console.log(e.target.value)}>
                Another List Element
              </LinkListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("All", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              <LinkListItem
                className="className"
                to="/"
                onClick={e => console.log(e.target.value)}
              >
                One ListElement
              </LinkListItem>
              <LinkListItem
                className="className"
                to="/"
                onClick={e => console.log(e.target.value)}
              >
                Another List Element
              </LinkListItem>
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
