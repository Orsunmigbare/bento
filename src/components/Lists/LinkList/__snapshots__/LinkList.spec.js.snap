// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkListItem Snapshots All 1`] = `
<ul
  className="sc-AykKF UQtjO"
>
  <li
    className="sc-AykKD cOQQmB className"
  >
    <a
      className="sc-AykKE jcvoKt"
      href="/"
      onClick={[Function]}
    >
      One ListElement
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
  <li
    className="sc-AykKD cOQQmB className"
  >
    <a
      className="sc-AykKE jcvoKt"
      href="/"
      onClick={[Function]}
    >
      Another List Element
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
</ul>
`;

exports[`LinkListItem Snapshots No props 1`] = `
<ul
  className="sc-AykKF UQtjO"
>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
    >
      One ListElement
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
    >
      Another List Element
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
</ul>
`;

exports[`LinkListItem Snapshots className 1`] = `
<ul
  className="sc-AykKF UQtjO"
>
  <li
    className="sc-AykKD cOQQmB className"
  >
    <a
      className="sc-AykKE jcvoKt"
    >
      One ListElement
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
  <li
    className="sc-AykKD cOQQmB className"
  >
    <a
      className="sc-AykKE jcvoKt"
    >
      Another List Element
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
</ul>
`;

exports[`LinkListItem Snapshots onClick 1`] = `
<ul
  className="sc-AykKF UQtjO"
>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
      onClick={[Function]}
    >
      One ListElement
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
      onClick={[Function]}
    >
      Another List Element
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
</ul>
`;

exports[`LinkListItem Snapshots to 1`] = `
<ul
  className="sc-AykKF UQtjO"
>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
      href="/"
    >
      One ListElement
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
  <li
    className="sc-AykKD cOQQmB"
  >
    <a
      className="sc-AykKE jcvoKt"
      href="/"
    >
      Another List Element
       
      <svg
        className="sc-AykKC corfaz"
        height="12"
        viewBox="0 0 20 33"
        width="7"
      >
        <path
          d="M12.15 16.5L.87 4.81c-.5-.52-.5-1.35.01-1.87L2.9.9a1.3 1.3 0 0 1 1.86 0l14.36 14.67a1.36 1.36 0 0 1 0 1.88L4.76 32.1a1.3 1.3 0 0 1-1.86.01L.88 30.06c-.5-.52-.5-1.34 0-1.87L12.14 16.5h.01z"
        />
      </svg>
    </a>
  </li>
</ul>
`;
